packages/flutter_tools/lib/src/build_system/targets/native_assets.dart

```diff
@@ -59,7 +60,7 @@ abstract class DartBuild extends Target {
             fileSystem,
             environment.logger,
             runPackageName,
-            includeDevDependencies: false,
+            includeDevDependencies: includeDevDependencies,
```

Comment:
Okay so the current way that dev dependencies works is that they should be included on all builds except release builds. I would image that the same should be true for native assets.

Instead of piping through this setting, could you just check here that the build mode (kBuildMode) is not release?

Reply:
This significantly simplifies the PR, thanks @vashworth! 🙏

All green locally, let's see if the CI is happy too.